src/components/VisualSkeletalAnimation.cpp

```diff
@@ -53,10 +53,14 @@ namespace REGoth
     deleteObjectSubtree();
     mModelScript = modelScript;
 
+    createAnimationMap();
+  }
+
+  void VisualSkeletalAnimation::createAnimationMap()
+  {
     for (bs::HAnimationClip clip : mModelScript->getAnimationClips())
```

Comment:
Shouldn't we make sure that `mModelScript` is valid in this function as it could be called from `onDeserializationEnded`?

Reply:
Good point. I'll put a safety-check.